src/Resources/app/administration/src/module/findologic-module/components/findologic-config/findologic-config.html.twig

```diff
@@ -7,58 +7,106 @@
                     {% block findologic_credentials_card_container %}
                         {% block findologic_credentials_settings %}
                             <div class="findologic-settings-credentials-fields">
-                                <sw-text-field
-                                        v-model="actualConfigData['FinSearch.config.shopkey']"
-                                        :label="$tc('findologic.settingForm.config.shopkey.label')"
-                                        :helpText="$tc('findologic.settingForm.config.shopkey.tooltipText')"
-                                        :required="true"
-                                        :error="shopkeyErrorState"
-                                >
-                                </sw-text-field>
-                                <sw-switch-field
+                                <div :class="disabledClass">
+                                    <sw-text-field
+                                            v-model="actualConfigData['FinSearch.config.shopkey']"
+                                            :label="$tc('findologic.settingForm.config.shopkey.label')"
+                                            :helpText="$tc('findologic.settingForm.config.shopkey.tooltipText')"
+                                            :required="true"
+                                            class="findologic-shopkey"
+                                            :disabled="!selectedLanguageId"
+                                            :placeholder="shopkeyPlaceholder"
+                                            :error="shopkeyErrorState"
+                                    >
+                                    </sw-text-field>
+                                </div>
+
+                                <sw-inherit-wrapper
                                         v-model="actualConfigData['FinSearch.config.active']"
-                                        :label="$tc('findologic.settingForm.config.active.label')"
-                                        :helpText="$tc('findologic.settingForm.config.active.tooltipText')"
+                                        :inheritedValue="selectedSalesChannelId == null ? null : allConfigs['null']['FinSearch.config.active']"
```

Comment:
1. Please be explicit by using `===`. If you are unsure if the variable is even set, you can also explicitly check for that as well by using `typeof`.
2. Move this logic to a helper method, since you do this check for basically all options.